### PR TITLE
Edit form

### PR DESCRIPTION
**Pull request by raphaeldieu**
fix sorting issues of the fields when delete / styling

### CODE REVIEW

File: app/assets/javascripts/angular/controller/forms/updateForms.js.erb [hunk 1]
```diff
@@ -71,27 +104,38 @@ app.controller('FormsUpdateCtrl', ['$scope', '$http', '$auth', '$location', '$ro
...
-  $scope.deleteOption = function(option,$index){
-    $scope.options.splice($index,1)
+  $scope.deleteOption = function(option){
+    $scope.options.splice($scope.options.indexOf(option),1);
   }
 
   //Update and hide options
   $scope.hideAndSaveEditPanel = function(field,labelInput){
-    
+    // var index = $scope.fields.indexOf(field)
```
Comment (f3r) [inline app/assets/javascripts/angular/controller/forms/updateForms.js.erb]: pls remove commented lines in PR

**Comment by h8rry** [comment]
@raphaeldieu please check

**Comment by raphaeldieu** [comment]
http://media.giphy.com/media/Jr7YlhsCbwZDG/giphy-facebook_s.jpg

**Comment by raphaeldieu** [comment]
Sooooooorry

**Comment by raphaeldieu** [comment]
on the merge conflicts now :)

**Comment by raphaeldieu** [comment]
@ggharry @f3r Done ;)

File: app/assets/templates/forms/update.html [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
         </div>
-        <div class="col-xs-4">
-          <h4>{{ name }}</h4>
-          <ul>
-            <li ng-repeat="field in fields track by $index" ng-mouseover="showButtons = true" ng-mouseleave="showButtons = false">
-              <button ng-click="delete(field,$index)" ng-show="!editPanelVisible && showButtons">X</button>
+
+        <div class="col-xs-6" id="listOfContents">
+          <ul class="contentField">
+            <li ng-repeat="field in fields | orderFieldBy:'index'" ng-mouseover="showButtons = true" ng-mouseleave="showButtons = false" class="field btn btn-embossed btn-primary">
```
Comment (h8rry) [inline app/assets/templates/forms/update.html:21]: `orderBy` doesn't work for you? Why do you need to make a custom filter function `orderFieldBy`?

Comment (raphaeldieu) [inline app/assets/templates/forms/update.html:21]: @ggharry  orderBy only works with arrays,
I want to order by 'index' which is not the order of an array but a value that we allocate (see schema.rb) to keep track of the position. So I need to filter by the key index.

Comment (h8rry) [inline app/assets/templates/forms/update.html:21]: Don't you have an array of objects? Or you have an object of objects?

Comment (h8rry) [inline app/assets/templates/forms/update.html:21]: You want to ng-repeat an array of `contents`, right?

Comment (h8rry) [inline app/assets/templates/forms/update.html:21]: http://justinklemm.com/angularjs-filter-ordering-objects-ngrepeat/

**Comment by raphaeldieu** [comment]
@ggharry @f3r  Is it ok ? 
Alt text

**Comment by h8rry** [comment]
I replied https://github.com/WDI-HK-8/Formidable-web/pull/24#discussion_r39335986

**Pull request merged**